test/node_stubs_test.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package test
+
+import (
+	"context"
+	"encoding/asn1"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"sync/atomic"
+	"testing"
+
+	"github.com/IBM/idemix/common/flogging"
+	smartbft_types "github.com/hyperledger-labs/SmartBFT/pkg/types"
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-protos-go-apiv2/orderer"
+	"github.com/hyperledger/fabric-x-orderer/common/types"
+	"github.com/hyperledger/fabric-x-orderer/common/utils"
+	"github.com/hyperledger/fabric-x-orderer/config"
+	node_util "github.com/hyperledger/fabric-x-orderer/node"
+	"github.com/hyperledger/fabric-x-orderer/node/comm"
+	"github.com/hyperledger/fabric-x-orderer/node/consensus/state"
+	protos "github.com/hyperledger/fabric-x-orderer/node/protos/comm"
+	"github.com/hyperledger/fabric-x-orderer/testutil"
+
+	"github.com/stretchr/testify/require"
+)
+
+type stubConsenter struct {
+	certificate []byte
+	key         []byte
+	server      *comm.GRPCServer // GRPCServer instance represents the consenter
+	txs         uint32           // Number of txs received from router
+	partyID     types.PartyID
+	decisions   chan *common.Block
+}
+
+func NewStubConsenter(t *testing.T, configStoreDir string, nodeConfigPath string, listener net.Listener) *stubConsenter {
```

Comment:
is it possible to reuse the stubs already in the router tests?
both consenter and batcher
maybe extend them somehow